reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -1759,7 +1759,9 @@ public T block(Duration timeout) {
 	 * @return a new {@link Mono}
 	 * @see #as as() for a loose conversion to an arbitrary type
 	 * @see #transform(Function)
+	 * @deprecated use {@link #as(Function)} with {@link #defer(Supplier)}
```

Comment:
should use transform, as is a full type conversion

Reply:
see my comment about `#transform` in the PR's description - not that trivial